packages/sdk/src/document/document.ts

```diff
@@ -1027,6 +1049,36 @@ export class Document<T, P extends Indexable = Indexable> {
     throw new YorkieError(Code.ErrInvalidArgument, `"${arg1}" is not a valid`);
   }
 
+  /**
+   * subscribeBroadcastEvent registers a callback to subscribe to broadcast events
+   * on the document. The callback will be called when the document receives a
+   * broadcast event with the given topic.
+   */
+  public subscribeBroadcastEvent(
```

Comment:
@hackerwins, @chacha912  I think we need to discuss whether to integrate this function with the `subscribe` function above by passing an event type (e.g., 'my-presence', 'broadcast') and a broadcast topic (or null) together (defining a new type using `pair`).

Reply:
I'm still unsure about the best approach for the subscribe interface. I've summarized the interfaces of other libraries and our current proposal:

* [Liveblocks](https://liveblocks.io/docs/api-reference/liveblocks-client#Room.subscribe.event)
   * Doesn't use separate topics, filters by event type within the event
   * The callback function receives `{event, user, connectionId}` as its argument, which includes information about the user who sent the event.

```js
// User 1
room.broadcastEvent({ type: "REACTION", emoji: "🔥" });

// User 2
const unsubscribe = room.subscribe("event", ({ event, user, connectionId }) => {
  //                                                  ^^^^ Will be User 1
  if (event.type === "REACTION") {
    // Do something
  }
});
```

* [Supabase](https://supabase.com/docs/guides/realtime/broadcast)
   * Callback function receives only `(payload)` as an argument
   * Subscribe to based on event topic.

```js
// User1
const channelB = supabase.channel('room-1')
channelB.send({
  type: 'broadcast',
  event: 'test',
  payload: { message: 'hello, world' },
})

// User2
const channelA = supabase.channel('room-1')
channelA
  .on(
    'broadcast',
    { event: 'test' }, // topic
    (payload) => messageReceived(payload)
  )
  .subscribe()
```

* Yorkie
   * Option 1
      ```js
      // User1
      const payload = 'hello';
      doc.broadcast('TOPIC_NAME', payload);
      
      // User2
      doc2.subscribeBroadcastEvent('TOPIC_NAME', (topic, payload) => {
          // Handle the broadcast event for the specified topic
      });
      ```
   * Option 2 (current)
      ```js
      // User1
      const payload = 'hello';
      doc.broadcast('TOPIC_NAME', payload);
      
      // User2
      doc2.subscribe({ type: 'broadcast', topic: 'TOPIC_NAME'}, (topic, payload) => {
          // Handle the broadcast event for the specified topic
      });
      ```

* Is there a reason for passing both `topic` and `payload` in the callback function?
* While Option 2 uses a common subscribe method, I'm concerned that adding the `BroadcastSubscribePair` type might have made it more complex.

Reply:
@chacha912 @sejongk @hackerwins 

I totally agree that `BroadcastSubscribePair` might be too complex to use. What do you think about aligning it with the existing event subscription pattern? The main reason I introduced `BroadcastSubscribePair` was to prevent users from having to compare topic values manually. However, after reviewing the subscription methods for other events, I noticed that users are already handling similar comparisons themselves.

```javascript
doc.subscribe('presence', (event) => {
              // Skipped
              } else if (event.type === 'unwatched') {
                cursors.removeCursor(event.value.clientID);
              } 
});
```

Therefore, I think it might be better to unify the subscription approach for broadcast events in the same way:

```javascript
 const broadcastTopic = 'test';
 const unsubscribe = d2.subscribe('broadcast', (topic, payload) => {
          if (topic === broadcastTopic) {
            eventCollector.add(payload);
          }
});
```

Of course, the potential issue with the interface mentioned above is that handlers could be triggered unintentionally if the comparison logic isn't implemented correctly. However, it also provides users with more flexibility. I'll go ahead and implement the new subscription interface in a commit—please review if it is okay.